File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlDataSize.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlDataSize : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.DataSize;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_DataSize(this);
    }

    public string Size { get; set; } = string.Empty;
    public int Scale { get; set; }
    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        if (!string.IsNullOrEmpty(Size))
        {
            sql.Write("(");
            sql.Write($"{Size}");
            if (Scale > 0)
            {
                sql.Write($", {Scale}");
            }
            sql.Write(")");
        }

        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlDataSize.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlDataSize implements ISqlExpression {
    SqlType: SqlType = SqlType.DataSize;
    Span: TextSpan = new TextSpan();
    Size: string = '';
    Scale: number = 0;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_DataSize(this);
    }

    ToSql(): string {
        if (!this.Size) return '';
        let sql = '(' + this.Size;
        if (this.Scale > 0) {
            sql += ', ' + this.Scale;
        }
        sql += ')';
        return sql;
    }
} 